automation/taskcluster/decision_task_nightly.py

```diff
@@ -3,95 +3,151 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 """
-Decision task for releases
+Decision task for nightly releases.
 """
 
-from __future__ import print_function
-import datetime
+import argparse
 import json
+import lib.tasks
 import os
 import taskcluster
 
-import lib.tasks
-
 TASK_ID = os.environ.get('TASK_ID')
+SCHEDULER_ID = os.environ.get('SCHEDULER_ID')
 HEAD_REV = os.environ.get('MOBILE_HEAD_REV')
 
-def generate_build_task():
-    created = datetime.datetime.now()
-    expires = taskcluster.fromNow('1 year')
-    deadline = taskcluster.fromNow('1 day')
-
-    command = "automation/taskcluster/actions/nightly.sh"
-
-    return {
-        "workerType": 'gecko-focus',
-        "taskGroupId": TASK_ID,
-        "expires": taskcluster.stringDate(expires),
-        "retries": 5,
-        "created": taskcluster.stringDate(created),
-        "tags": {},
-        "priority": "lowest",
-        "schedulerId": "focus-nightly-sched",
-        "deadline": taskcluster.stringDate(deadline),
-        "dependencies": [ TASK_ID ],
-        "routes": [
-            "index.project.mobile.reference-browser.nightly.latest"
-        ],
-        "scopes": [
-            "queue:route:index.project.mobile.reference-browser.nightly.*",
-            "secrets:get:project/mobile/reference-browser/preview-key-store",
+BUILDER = lib.tasks.TaskBuilder(
+    task_id=TASK_ID,
+    owner="skaspari@mozilla.com",
+    source='https://github.com/mozilla-mobile/reference-browser/raw/{}/.taskcluster.yml'.format(HEAD_REV),
+    scheduler_id=SCHEDULER_ID
+)
+
+
+def generate_build_task(apks):
+    artifacts = {}
+    for apk in apks:
+        artifact = {
+            "type": 'file',
+            "path": apk,
+            "expires": taskcluster.stringDate(taskcluster.fromNow('1 year'))
+        }
+        artifacts["public/%s" % os.path.basename(apk)] = artifact
+
+    checkout = "git fetch origin && git reset --hard origin/master"
+    assemble_task = 'assembleRelease'
+
+    return taskcluster.slugId(), BUILDER.build_task(
+        name="(Reference Browser) Build task",
+        description="Build Reference Browser from source code.",
+        command=(checkout +
+                 ' && python automation/taskcluster/helper/get-secret.py '
+                 '-s project/mobile/reference-browser/sentry -k dsn -f .sentry_token'
+                 ' && ./gradlew --no-daemon clean test ' + assemble_task),
+        features={
+            "chainOfTrust": True   # TODO need 'taskClusterProxy': True?
```

Comment:
`taskclusterProxy` is needed if you access TC Secrets (like this one https://tools.taskcluster.net/secrets/project%2Fmobile%2Freference-browser%2Fsentry). Feel free to try without it => this may give you more context about what this proxy does :slightly_smiling_face: